### PR TITLE
[Enhancement](log) improve error msg for delta writer fail (#12121)

### LINKED ISSUE #12121: [Enhancement] need detailed error message when delta writer fail to write
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Current warning log contains hardly anything useful to diagnose why write fails:
```
//be/src/runtime/tablets_channel.cpp
//Status TabletsChannel::add_batch

    auto err_msg = strings::Substitute(
                   "tablet writer write failed, tablet_id=$0, txn_id=$1, err=$2",
                   tablet_to_rowidxs_it.first, _txn_id, st.code());
```
The `st.code()` is too generic to identify the root cause. E.g., when I got err=6, aka INTERNAL_ERROR, it could be OLAP_ERR_ALREADY_CANCELLED, OLAP_ERR_TOO_MANY_SEGMENTS, OLAP_ERR_TABLE_NOT_FOUND, OLAP_ERR_TOO_MANY_VERSION, etc.

Althougn we can use 'side-channel' information to guess what has happened, I think it better to offer a detailed error message directly.

### Solution

privide info like:
```
st.precise_code();
st.get_error_msg();
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by freemandealer**
Signed-off-by: freemandealer <freeman.zhang1992@gmail.com>

# Proposed changes

Issue Number: close #12121

## Problem summary
fuzzy error msg when delta writer writing failed

Describe your changes.
add error code and err stack for detailed info.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [x] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [x] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [x] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by stalary** [comment]
It need be code format.

**Comment by freemandealer** [comment]
> It need be code format.

sure, will do.

File: be/src/runtime/tablets_channel.cpp [hunk 1]
```diff
@@ -375,8 +375,10 @@ Status TabletsChannel::add_batch(const TabletWriterAddRequest& request,
         Status st = tablet_writer_it->second->write(&send_data, tablet_to_rowidxs_it.second);
         if (!st.ok()) {
             auto err_msg = strings::Substitute(
-                    "tablet writer write failed, tablet_id=$0, txn_id=$1, err=$2",
-                    tablet_to_rowidxs_it.first, _txn_id, st.code());
+                    "tablet writer write failed, tablet_id=$0, txn_id=$1, err=$2"
+                    ", errcode=$3, msg:$4",
```
Comment (zhannngchen) [inline be/src/runtime/tablets_channel.cpp:379]: errcode -> precise_code might be better?

**Review by zhannngchen — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
This PR could not be picked because 1.1.2 do not have this method.

**Pull request merged**